packages/host/app/services/matrix-service.ts

```diff
@@ -228,13 +229,96 @@ export default class MatrixService extends Service {
       try {
         await this._client.startClient();
         await this.initializeRooms();
+
+        // TODO this is a temporary measure to prove that we can obtain a realm session.
+        // ultimately we need to figure out a better approach in terms of when/where we
+        // obtain sessions for realms that we care about.
+        await this.getRealmToken(new URL(ownRealmURL));
       } catch (e) {
         console.log('Error starting Matrix client', e);
         await this.logout();
       }
     }
   }
 
+  async getRealmToken(
```

Comment:
I opted to put this in the matrix-service because we initiate it as soon as we have a matrix session, and this depends on matrix service methods. Adding this to it's own service would create a circular dependency on the matrix service....